File: LR/LR_7-8/2.py
text = str(input("Текст: "))
plus_cnt = 0
minus_cnt = 0
plus_zero_cnt = 0
minus_zero_cnt = 0

for i in range(len(text)):
    if (text[i] == "+"):
        plus_cnt += 1
        if ((i+1) <= len(text) and text[i+1] == "0"):
            plus_zero_cnt += 1

print("-: " + str(minus_cnt))
print("+: " + str(plus_cnt))
print("-0: " + str(minus_zero_cnt))
print("+0: " + str(plus_zero_cnt))

File: LR/LR_7-8/2.pas
var
  text: string;
  i, plus_cnt, minus_cnt, plus_zero_cnt, minus_zero_cnt: Integer;
begin
  Write('Текст: ');
  ReadLn(text);
  
  plus_cnt := 0;
  minus_cnt := 0;
  plus_zero_cnt := 0;
  minus_zero_cnt := 0;

  for i := 1 to Length(text) do
  begin
    if text[i] = '+' then
    begin
      plus_cnt := plus_cnt + 1;
      if (i < Length(text)) and (text[i + 1] = '0') then
        plus_zero_cnt := plus_zero_cnt + 1;
    end
    else if text[i] = '-' then
    begin
      minus_cnt := minus_cnt + 1;
      if (i < Length(text)) and (text[i + 1] = '0') then
        minus_zero_cnt := minus_zero_cnt + 1;
    end;
  end;

  WriteLn('-: ', minus_cnt);
  WriteLn('+: ', plus_cnt);
  WriteLn('-0: ', minus_zero_cnt);
  WriteLn('+0: ', plus_zero_cnt);
end.